### DISCUSSION TITLE
Delete Last Spawned Vehicle Help

### DISCUSSION THREAD

**Discussion by SlavicBlin** [YimMenu/YimMenu#2512] [Q&A]
I have tried multiple methods to try and get this working however it just cant seem to delete the last drive vehicle.
i can easily get it to do
void delete_previous_spawn(Vehicle previousVehicle)
{
    if (ENTITY::DOES_ENTITY_EXIST(previousVehicle) && ENTITY::IS_ENTITY_A_VEHICLE(previousVehicle))
    {
        VEHICLE::DELETE_VEHICLE(&previousVehicle);
    }
}
but when adding a GET_LAST_DRIVEN_VEHICLE it just does not want to delete the last driven vehicle
i have tried this
void delete_previous_spawn(Vehicle newVehicle)
{
    // Get the last driven vehicle
    Vehicle lastVehicle = VEHICLE::GET_LAST_DRIVEN_VEHICLE();

    // Check if the last vehicle exists and is different from the new vehicle
    if (ENTITY::DOES_ENTITY_EXIST(lastVehicle) && ENTITY::IS_ENTITY_A_VEHICLE(lastVehicle) && lastVehicle != newVehicle)
    {
        // Delete the last vehicle
        VEHICLE::DELETE_VEHICLE(&lastVehicle);
    }
}
but it also did not work.
previously i used " if (spawn_vehicle.spawn_last) { Vehicle veh = VEHICLE::GET_LAST_DRIVEN_VEHICLE(); VEHICLE::DELETE_VEHICLE(&veh); } " as suggested by [takoyakiwhite] but in the latest builds this has not worked :(

Iam still very new to coding in general so any help would be highly appriciated 👍

**Comment by SlavicBlin** [comment]
Solved using the help of @tupoy-ya  <3

**Reply by pierrelasse** [reply]
Could you maybe provide the solution for others that are might searching for it?

**Reply by SlavicBlin** [reply]
can do

**Reply by pierrelasse** [reply]
Cool

**Comment by SlavicBlin** [comment]
Please note **Iam new to coding** so if this doesn't make sense iam sorry but i have provided all these files in my repo here https://github.com/SlavicBlin/Delete-Last-Vehicle  i would not recommened copy and pasting since i have changed a few things in the menu to suite my needs so you might loose features and or cause crashes in your menu.


Step by step of how to implement into your own build of both Yimmenu and TupoyeMenu ( I currently use TupoyeMenu ) so it might differ slightly
Go to `view_spawn_vehicle.cpp` and add the following lines of code. 
`ImGui::SameLine();
components::command_checkbox<"spawnlast">();`
i would add this under `components::command_checkbox<"spawnmaxed">();`
then add 
`if (g.spawn_vehicle.spawn_last)
{
 vehicle::delete_last_spawned(veh);
}` under the following
 `if (g.spawn_vehicle.spawn_inside)
{
vehicle::teleport_into_vehicle(veh);
}`
repeat for all areas in `view_spawn_vehicle.cpp` that mention spawning maxed or spawning inside.

Then head to `settings.hpp` and find `struct spawn_vehicle` and below where it says `bool spawn_maxed   = false;` add a line that says
`bool spawn_last  = false;` then look for `NLOHMANN_DEFINE_TYPE_INTRUSIVE(spawn_vehicle` and add `spawn_last` in it, should look
like this 
`NLOHMANN_DEFINE_TYPE_INTRUSIVE(spawn_vehicle, preview_vehicle, spawn_inside, spawn_last, spawn_maxed, plate, spawn_type)
} spawn_vehicle{};`
		 

	
Next! head over to `spawn_vehicle.cpp` add the following lines 
`if (g.spawn_vehicle.spawn_last)
	{
		vehicle::delete_last_spawned(veh);
	}`
This goes under 
`if (g.spawn_vehicle.spawn_inside && ctx->get_sender()->id() == self::id)
	{
		vehicle::teleport_into_vehicle(veh);
`
after that near the bottom you should see bool commands.
add this line at the bottom of thoes commands.
`bool_command g_spawn_last("spawnlast", "Delete Last", "Deletes the last spawned vehicle",
    g.spawn_vehicle.spawn_last);
`

Next step!
Head to `vehicle.hpp` and find 
`namespace big::vehicle
{
	struct disable_collisions
	{
		inline static memory::byte_patch* m_patch;
	};
`
it should be right at the top and will contain a bunch of floats and bools and stuff anyways locate
 `void teleport_into_vehicle(Vehicle veh);` and underneath it add the following
 `void delete_last_spawned(Vehicle veh);`
it should look exactly how it does on here with tp above and delete below 
 
 Next find inside of `vehicle.cpp`
`void teleport_into_vehicle(Vehicle veh)
	{
		PED::SET_PED_INTO_VEHICLE(self::ped, veh, -1);
	}`
	once you do that add the following line underneath it 
	`void delete_last_spawned(Vehicle veh)
	{
		Vehicle lastspawned = PLAYER::GET_PLAYERS_LAST_VEHICLE();
		 entity::delete_entity(lastspawned);
	}
`

EDIT: Forgot to add settings.hpp

**Comment by SlavicBlin** [comment]
also here is the discussion in Tupoye's git https://github.com/TupoyeMenu/TupoyeMenu/discussions/85
